arviz/stats/stats.py

```diff
@@ -349,84 +351,92 @@ def hpd(ary, credible_interval=None, circular=False, multimodal=False, skipna=Fa
         if not 1 >= credible_interval > 0:
             raise ValueError("The value of credible_interval should be in the interval (0, 1]")
 
-    if ary.ndim > 1:
-        hpd_array = np.array(
-            [
-                hpd(
-                    row,
-                    credible_interval=credible_interval,
-                    circular=circular,
-                    multimodal=multimodal,
-                )
-                for row in ary.T
-            ]
-        )
-        return hpd_array
-
     if multimodal:
-        if skipna:
-            ary = ary[~np.isnan(ary)]
-
-        if ary.dtype.kind == "f":
-            density, lower, upper = _fast_kde(ary)
-            range_x = upper - lower
-            dx = range_x / len(density)
-            bins = np.linspace(lower, upper, len(density))
-        else:
-            bins = get_bins(ary)
-            _, density, _ = histogram(ary, bins=bins)
-            dx = np.diff(bins)[0]
+        return _hpd_multimodal(ary, credible_interval, skipna)
+
+    if isinstance(ary, np.ndarray):
+        ary = convert_to_dataset(ary)
```

Comment:
If input is array, output should be array too (like wrap_xarray_ufunc does).

We have to decide what should happen though, do we want `hpd` to be backwards compatible when input is ndarray? Or should we assume ArviZ chain, draw, *shape convention? @aloctavodia

Reply:
backwards compatibility when input is ndarray sounds reasonable.

Reply:
Okay, I'll do it.